### PR TITLE
Fix sourcemaps support with TS

### LINKED ISSUE #7811: Go to definition doesn't work in VSCode
Following #7580

The "go to definition" VSCode function used to work for react-admin (at least since #6039). But in v4, we've [switched from `tsc` to `tsup`](https://github.com/marmelab/react-admin/pull/6993), and the `declarationMap` directive in `tsconfig.json` isn't honored anymore. This seems to be a limitation of `tsup` (cf https://github.com/egoist/tsup/issues/488), and a serious regression.

The only solution seems to go back to `tsc` until `tsup` does a better job.

### PR DESCRIPTION
**Pull request by djhi**
Fixes #7811, #7580

Reverts #6993

### CODE REVIEW

File: packages/ra-core/package.json [hunk 1]
```diff
@@ -20,8 +20,10 @@
     "bugs": "https://github.com/marmelab/react-admin/issues",
     "license": "MIT",
     "scripts": {
-        "build": "tsup --external react-hook-form src/index.ts --silent --clean --format cjs,esm --minify --keep-names --metafile --sourcemap --dts --legacy-output",
-        "watch": "tsup --external react-hook-form src/index.ts --silent --clean --format cjs,esm --minify --keep-names --metafile --sourcemap --dts --legacy-output --watch"
+        "build": "yarn run build-cjs && yarn run build-esm",
+        "build-cjs": "rimraf ./dist/cjs && tsc --outDir dist/cjs",
+        "build-esm": "rimraf ./dist/esm && tsc --outDir dist/esm --module es2015",
+        "watch": "tsc --outDir esm --module es2015 --watch"
```
Comment (slax57) [inline packages/ra-core/package.json:26]: Really not an expert of this stuff, but, shouldn't it be like this?
```suggestion
        "watch": "tsc --outDir dist/esm --module es2015 --watch"
```

**Review by slax57 — APPROVED** [review]

**Comment by fzaninotto** [comment]
When comparing with the PR this is reverting (#6993), I see many small differences. Are there on purpose?

**Comment by djhi** [comment]
> When comparing with the PR this is reverting (https://github.com/marmelab/react-admin/pull/6993), I see many small differences. Are they on purpose?

Can you ellaborate ?

**Comment by fzaninotto** [comment]
in 3.0:

```
    "main": "lib/index",
    "module": "esm/index.js",
    "types": "esm/index.d.ts",
```

your PR:

```
    "main": "dist/cjs/index.js",
    "module": "dist/esm/index.js",
    "types": "dist/cjs/index.d.ts",
```

in 3.0:

```
        "build": "yarn run build-cjs && yarn run build-esm",
        "build-cjs": "rimraf ./lib && tsc",
        "build-esm": "rimraf ./esm && tsc --outDir esm --module es2015",
        "watch": "tsc --outDir esm --module es2015 --watch"
```

Your PR:
```
        "build": "yarn run build-cjs && yarn run build-esm",
        "build-cjs": "rimraf ./dist/cjs && tsc --outDir dist/cjs",
        "build-esm": "rimraf ./dist/esm && tsc --outDir dist/esm --module es2015",
        "watch": "tsc --outDir dist/esm --module es2015 --watch"
```

My question is: why these differences? Since the 3.0 setup was known to work, why didn't you use it as is?

**Comment by djhi** [comment]
Just to have a single output directory (`./dist`)

**Pull request merged**